services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -175,13 +170,16 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
     } yield {
       (projectVersions, fileCommits) match {
         case (Right(tagsProject), Right(tagsFiles)) =>
-          Right(for {
-            tagProject <- tagsProject
-            tagFile <- tagsFiles
-            if tagFile.commitId == tagProject.commit.id
-          } yield tagProject)
-        case (_, Left(exception)) => Left(VersioningException(exception.message))
-        case (Left(exception), _) => Left(VersioningException(exception.message))
+          Right(
+            tagsProject.flatMap(
+              tagProject =>
+                tagsFiles.withFilter(tagFile => tagFile.commitId == tagProject.commit.id).map(tagFile => tagProject)
+            )
```

Comment:
Are you sure that it's better than it was? :)
Maybe it's just me, but this code is much harder to read compared to its version on the right. @ilyaonishenko what do you think?

Reply:
I like maps. But for sure for-c is better here. Done.